### PR TITLE
Refactor/manual rubocop fixes part1

### PR DESCRIPTION
**Pull request by luke-hill**
# Description

Attempt to start fixing some more complex things. Likely to break things

## Type of change

Please delete options that are not relevant.

- Refactoring (improvements to code design or tooling that don't change behaviour)
- Breaking change (will cause existing functionality to not
  work as expected) _Maybeeeeeee_

Please add an entry to the relevant section of CHANGELOG.md as part of this pull request. - TODO

## Note to other contributors

If your change may impact future contributors, explain it here, and remember to update README.md and CONTRIBUTING.md accordingly.

# Checklist:

Your PR is ready for review once the following checklist is
complete. You can also add some checks if you want to.

- [ ] Tests have been added for any changes to behaviour of the code
- [ ] New and existing tests are passing locally and on CI
- [ ] `bundle exec rubocop` reports no offenses
- [ ] RDoc comments have been updated
- [ ] CHANGELOG.md has been updated

### CODE REVIEW

**Comment by luke-hill** [comment]
Hi @mattwynne 

I've edited a few `lib/` files here and It got quite confusing, so for a lot of it I opted to do as little as possible given I don't know an awful lot about the areas. If you could review each of those it would be greatly appreciated.

**Comment by mattwynne** [comment]
There's a lot here to review @luke-hill! I'd say if the tests are passing you're probably good.

Was there anything specific you want me to look at in detail?

File: lib/cucumber/core/gherkin/parser.rb [hunk 1]
```diff
@@ -18,19 +18,8 @@ def initialize(receiver, event_bus, gherkin_query)
...
-            if !message.gherkin_document.nil?
-              event_bus.gherkin_source_parsed(message.gherkin_document)
-            elsif !message.pickle.nil?
-              receiver.pickle(message.pickle)
-            elsif message.parse_error
-              raise Core::Gherkin::ParseError.new("#{document.uri}: #{message.parse_error.message}")
-            else
-              raise "Unknown message: #{message.to_hash}"
-            end
+          source_messages(document).each do |message|
```
Comment (luke-hill) [inline lib/cucumber/core/gherkin/parser.rb:21]: This method (and new submethods) wrt naming / partitioning (Item 1 of 3)

File: lib/cucumber/core/report/summary.rb [hunk 2]
```diff
@@ -20,19 +20,27 @@ def ok?(be_strict = Test::Result::StrictConfiguration.new)
         private
 
         def subscribe_to(event_bus)
+          register_test_case_finished_listener(event_bus)
```
Comment (luke-hill) [inline lib/cucumber/core/report/summary.rb:23]: This methods wrt naming / partioning (and submethods), part 2 of 3

File: lib/cucumber/core/test/result.rb [hunk 3]
```diff
@@ -374,7 +374,7 @@ def total(for_status = nil)
             if for_status
               @totals.fetch(for_status, 0)
             else
-              @totals.reduce(0) { |total, status| total += status[1] }
+              @totals.values.reduce(0) { |total,status| total + status }
```
Comment (luke-hill) [inline lib/cucumber/core/test/result.rb:377]: Small functional tweak here. Part 3 of 3

**Review by luke-hill — COMMENTED** [review]
I've reduced down the scope to 3 files / methods. Think that should cover 99% of the risk evaluation.

File: lib/cucumber/core/gherkin/parser.rb [hunk 4]
```diff
@@ -18,19 +18,8 @@ def initialize(receiver, event_bus, gherkin_query)
...
+            store_metadata(message, document)
```
Comment (mattwynne) [inline lib/cucumber/core/gherkin/parser.rb:22]:
```suggestion
            process(message, document)
```

Comment (mattwynne) [reply lib/cucumber/core/gherkin/parser.rb:22]: I think `store_metadata` is a bit specific - we do a bunch of things in this message, but in general we're processing that message, so I think that's a better name.

File: lib/cucumber/core/gherkin/parser.rb [hunk 5]
```diff
@@ -47,6 +36,27 @@ def done
...
           self
         end
+
+        private
+
+        def source_messages(document)
+          ::Gherkin.from_source(document.uri, document.body, gherkin_options(document))
+        end
+
+        def store_metadata(message, document)
```
Comment (mattwynne) [inline lib/cucumber/core/gherkin/parser.rb:46]:
```suggestion
        def process(message, document)
```

File: lib/cucumber/core/test/result.rb (hunk 3 above)
Comment (mattwynne) [reply lib/cucumber/core/test/result.rb:377]: I think `status` is the wrong name for the block param here. It's the count of scenarios with the given status, right?

```suggestion
              @totals.values.reduce(0) { |total,count| total + count }
```

**Pull request merged**